packages/@aws-cdk/assertions/lib/match.ts

```diff
@@ -265,6 +273,42 @@ class ObjectMatch extends Matcher {
   }
 }
 
+class SerializedJson extends Matcher {
+  constructor(
+    public readonly name: string,
+    private readonly pattern: any,
+  ) {
+    super();
+  };
+
+  public test(actual: any): MatchResult {
+    const result = new MatchResult(actual);
+    if (getType(actual) !== 'string') {
+      result.push(this, [], `Expected JSON as a string but found ${getType(actual)}`);
+      return result;
+    }
+    let parsed;
+    try {
+      parsed = JSON.parse(actual);
+    } catch (err) {
+      if (err instanceof SyntaxError) {
+        result.push(this, [], 'Invalid JSON string');
+        return result;
+      } else {
+        throw err;
+      }
+    }
+
+    if (parsed == null) {
+      result.push(this, [], 'Invalid JSON string');
+    }
+    const matcher = Matcher.isMatcher(this.pattern) ? this.pattern : new LiteralMatch(this.name, this.pattern);
```

Comment:
we should probably have a helper function for this :)

Reply:
I've thought about it, but there's no nice way to abstract it out. 
At each point, this logic needs to be thought about, such as like this - https://github.com/aws/aws-cdk/blob/a7320786f57d891bb74bfbc17a2169b65620006e/packages/%40aws-cdk/assertions/lib/match.ts#L257-L259.

Not providing a helper method forces this thinking, even though it feels like it's code duplication.

Happy to be told otherwise. I don't mind making a local function that does this boilerplate, but I don't feel it's "just" boilerplate.